GlobalstatsIO.cs

```diff
@@ -5,318 +5,315 @@
-		headers.Add("Cache-Control", "no-cache");
-		headers.Add("Content-Length", json_payload.Length.ToString());
-
-		byte[] pData = Encoding.ASCII.GetBytes(json_payload.ToCharArray());
-
-
-		WWW www = new WWW(url, pData, headers);
-
-
-		while (!www.isDone)
-		{
-			System.Threading.Thread.Sleep(100);
-		}
-
-		// check for errors
-		if (www.error == null)
-		{
-			//UnityEngine.Debug.Log ("WWW POST Ok!");
-		}
-		else
-		{
-			UnityEngine.Debug.Log("WWW POST Error: " + www.error);
-			UnityEngine.Debug.Log("WWW POST Content: " + www.text);
-			return new GlobalstatsIO_Leaderboard();
-		}
-
-		GlobalstatsIO_Leaderboard leaderboard = JsonUtility.FromJson<GlobalstatsIO_Leaderboard>(www.text);
-
-		return leaderboard;
-	}
-	
-}
+    [Serializable]
+    public class StatisticValues
+    {
+        public string key = null;
+        public string value = "0";
+        public string sorting = null;
+        public string rank = "0";
+        public string value_change = "0";
+        public string rank_change = "0";
+    }
+
+    [Serializable]
+    public class LinkData
+    {
+        public string url = null;
+        public string pin = null;
+    }
+
+    [Serializable]
+    public class LeaderboardValue
+    {
+        public string name = null;
+        public string user_profile = null;
+        public string user_icon = null;
+        public string rank = "0";
+        public string value = "0";
+    }
+
+    [Serializable]
+    public class Leaderboard
+    {
+        public LeaderboardValue[] data;
+    }
+
+    public class GlobalstatsIOClient
+    {
+        [HideInInspector]
+        public static string api_id = "";
+
+        [HideInInspector]
+        public static string api_secret = "";
+
+        [Serializable]
+        private class AccessToken
+        {
+            public string access_token = null;
+            public string token_type = null;
+            public string expires_in = null;
+            public int created_at = (int)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;
+
+            public bool IsValid()
+            {
+                //Check if still valid, allow a 2 minute grace period
+                return (created_at + int.Parse(expires_in) - 120) > (int)DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1)).TotalSeconds;
+            }
+        }
+
+        [Serializable]
+        private class StatisticResponse
+        {
+            public string name = null;
+            public string _id = null;
+
+            [SerializeField]
+            public List<StatisticValues> values = null;
+        }
+
+        private AccessToken api_access_token = null;
+
+        private List<StatisticValues> statistic_values = new List<StatisticValues>();
+
+        [HideInInspector]
+        public string statistic_id = "";
+
+        [HideInInspector]
+        public string user_name = "";
+
+        [HideInInspector]
+        public LinkData link_data = null;
```

Comment:
Would you mind moving those fields up to the beginning of the class?
Right now, field definitions are mixed between the private classes which makes it harder to read.